src/pages/main/MainPage.tsx

```diff
@@ -65,34 +81,53 @@ const MainPage: React.FC = () => {
   const onCloseModal = useCallback(() => {
     setModeData({ mode: MainPageModes.OVERVIEW, selectedMovie: null });
   }, []);
+
+  const navigateToMainPage = useCallback(() => {
+    history.push("/");
+  }, []);
+
+  const navigateToSearchPage = useCallback(() => {
+    history.push("/search?Search%20Query=");
+  }, []);
+
+  const setSearchingValue = useCallback((searchingValue) => {
+    history.push("/search?Search%20Query=" + searchingValue);
+    scrollToElement(movieBoard.current);
+  }, [])
   
   const filteredMovies = useMemo(() => filterMoviesBySearchingValue(movies, searchingValue), [searchingValue, movies]);
 
+  const MovieModal = useMemo(() => renderModal(mode), [mode]);
+  
   const contentStyles = mode !== MainPageModes.OVERVIEW ? bluredStyles: {};
 
   return (
     <>
       <header style={contentStyles}>
         <Banner
+          movieInOverview={movieInOverview}
           onChangeSearchingValue={setSearchingValue}
           onCreateMovie={onCreateMovie}
+          onLogoClicked={navigateToMainPage}
+          onSearchIconClicked={navigateToSearchPage}
```

Comment:
i believe this component should be divided into two different components with some 1 common wrapper.